pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -422,8 +422,9 @@ func (c *Controller) deleteService(svc *model.Service) {
 	shard := model.ShardKeyFromRegistry(c)
 	event := model.EventDelete
 	c.opts.XDSUpdater.SvcUpdate(shard, string(svc.Hostname), svc.Attributes.Namespace, event)
-
-	c.handlers.NotifyServiceHandlers(nil, svc, event)
+	if !svc.Attributes.ExportTo.Contains(visibility.None) {
```

Comment:
Why not filterout ealier? When it enter this function or ealier

Reply:
I just kept it at the endpoint to give a chance to make "SvcUpdate" call for any metric etc and to clean up service maps (by any chance if it is there)